src/Featurizers/UnitTests/TfidfVectorizerFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,190 @@
+// ----------------------------------------------------------------------
```

Comment:
Make sure that there are automated tests that cover all scenarios; you mentioned that tests were a work in progress but I want to make sure there are more cases than what you have here.